src/text/text.rs

```diff
@@ -580,6 +586,36 @@ where
     }
 }
 
+/// A trait for converting a value to a [`Text`].
+///
+/// This trait is automatically implemented for any type that implements the [`Display`] trait. As
+/// such, `ToText` shouldn't be implemented directly: [`Display`] should be implemented instead, and
+/// you get the `ToText` implementation for free.
+///
+/// [`Display`]: std::fmt::Display
+pub trait ToText<'a> {
+    fn to_text(&self) -> Text<'a>;
+}
+
+/// # Panics
+///
+/// In this implementation, the `to_text` method panics if the `Display` implementation returns an
+/// error. This indicates an incorrect `Display` implementation since `fmt::Write for String` never
+/// returns an error itself.
+impl<'a, T: fmt::Display> ToText<'a> for T {
+    fn to_text(&self) -> Text<'a> {
+        let lines = self
+            .to_string()
+            .lines()
+            .map(|line| Line::from(line.to_string()))
+            .collect();
+        Text {
+            lines,
+            ..Default::default()
+        }
+    }
+}
```

Comment:
Can this call one of the text constructors instead of duplicating it? Otherwise this may lead to divergent implementations in future.

Reply:
d3f786567fa6de51a02600a03c9b509823caa9aa